### ISSUE TITLE
bug: etcd keepalive function runtime error

### ISSUE THREAD

**Issue by zuiyangqingzhou** [apache/apisix#6887]
### Current Behavior

server-info plugin  error occurred when report info 

### Error Logs

```
2022/04/20 03:33:58 [warn] 17703#17703: *757946 [lua] v3.lua:151: keepalive(): invalid response code: 503. Retrying, context: ngx.timer
2022/04/20 03:34:06 [warn] 17703#17703: *757946 [lua] v3.lua:151: keepalive(): invalid response code: 503. Retrying, context: ngx.timer
2022/04/20 03:34:14 [warn] 17703#17703: *757946 [lua] v3.lua:151: keepalive(): invalid response code: 503. Retrying, context: ngx.timer
2022/04/20 03:34:22 [warn] 17703#17703: *757946 [lua] v3.lua:151: keepalive(): invalid response code: 503. Retrying, context: ngx.timer
2022/04/20 03:34:22 [error] 17703#17703: *757946 lua user thread aborted: runtime error: ...er_v2.1.0_20220420//deps/share/lua/5.1/resty/etcd/v3.lua:158: attempt to index local 'res' (a nil value)
stack traceback:
coroutine 0:
	...er_v2.1.0_20220420//deps/share/lua/5.1/resty/etcd/v3.lua: in function 'keepalive'
	...ffic-gateway/master_v2.1.0_20220420/apisix/core/etcd.lua:370: in function 'keepalive'
	...ay/master_v2.1.0_20220420/apisix/plugins/server-info.lua:217: in function 'report'
	...ay/master_v2.1.0_20220420/apisix/plugins/server-info.lua:279: in function <...ay/master_v2.1.0_20220420/apisix/plugins/server-info.lua:274>
coroutine 1:
	[C]: in function 'thread_wait'
	...traffic-gateway/master_v2.1.0_20220420/apisix/timers.lua:52: in function <...traffic-gateway/master_v2.1.0_20220420/apisix/timers.lua:32>
	[C]: in function 'pcall'
	...fic-gateway/master_v2.1.0_20220420/apisix/core/timer.lua:35: in function <...fic-gateway/master_v2.1.0_20220420/apisix/core/timer.lua:31>
	[C]: in function 'pcall'
	...fic-gateway/master_v2.1.0_20220420/apisix/core/timer.lua:58: in function <...fic-gateway/master_v2.1.0_20220420/apisix/core/timer.lua:51>, context: ngx.timer
2022/04/20 03:34:22 [error] 17703#17703: *757946 [lua] timer.lua:37: failed to run the timer: background err: ...traffic-gateway/master_v2.1.0_20220420/apisix/timers.lua:54: bad argument #2 to 'error' (expected table to have __tostring metamethod), context: ngx.timer
```

### Steps to Reproduce

none

### Environment

- APISIX version (run `apisix version`):  2.10.5
- Operating system (run `uname -a`):   x86_64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): nginx version: openresty/1.19.9.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): 3.5.0
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by spacewander** [comment]
Why do you think upgrading the etcd library can solve this problem?

**Comment by spacewander** [comment]
Look like it is solved in https://github.com/api7/lua-resty-etcd/commit/f75f4c551b048e00233df38ab725b46218b35de9, which is under 1.6.1

**Comment by zuiyangqingzhou** [comment]
> Look like it is solved in [api7/lua-resty-etcd@f75f4c5](https://github.com/api7/lua-resty-etcd/commit/f75f4c551b048e00233df38ab725b46218b35de9), which is under 1.6.1

Can version 2.10 be solved by directly upgrading lua-resty-etcd to 1.6.1?  

Are there compatibility issues?
